[build] - Updates for ROCm 6.2 sources

estewart08 opened the pull request:
Introduction of new component rocprofiler-register used in rocr, roct, and hipamd builds.

estewart08 committed:
[build] - Updates for ROCm 6.2 sources

ronlieb commented on bin/aomp_common_vars:
```diff
@@ -351,6 +351,7 @@ AOMP_ROCMCMAKE_REPO_NAME=${AOMP_ROCMCMAKE_REPO_NAME:-rocm-cmake}
 AOMP_GDB_REPO_NAME=${AOMP_GDB_REPO_NAME:-ROCgdb}
 AOMP_DBGAPI_REPO_NAME=${AOMP_DBGAPI_REPO_NAME:-ROCdbgapi}
 AOMP_PROF_REPO_NAME=${AOMP_PROF_REPO_NAME:-rocprofiler}
+AOMP_PROF_REGISTER_REPO_NAME=${AOMP_PROF_REGISTER_REPO_NAME:-rocprofiler-register}
```
missing the build script for rocprofiler-register

estewart08 replied:
Yep, accidentally left that off the commit. Added it back.

ronlieb reviewed (commented):
other than rocprofiler-register component, built fine and tested ok

estewart08 committed:
[build] - Add build_rocprofiler-register.sh

estewart08 commented:
> other than rocprofiler-register component, built fine and tested ok

Interesting, hipamd failed to build for me without rocprofiler-register.

estewart08 committed:
[build] - Update build_rocprofiler-register name in header

estewart08 committed:
[build] - Additional changes for 6.2 source move

The build of opencl is now dependent on opencl-icd-loader.
This used to be built alongside opencl, but has been deprecated.
The opencl-icd-loader package is built before opencl now, but
has no public sources at the moment. For now we will download
the prebuilt package and store it at /home/estewart/local.

estewart08 commented:
I also added opencl-icd-header to build_prereq.sh as the opencl build is dependent on it. This may be removed in favor of using the system package `ocl-icd-opencl-dev`
